### PR TITLE
Disable GFNI tests for NAOT

### LINKED ISSUE #110240: JIT.HardwareIntrinsics failing on native AOT
Not clear how we ended up generating unsupported instructions, this test should only be allowed to use x64 baseline instructions.

Seen in #110238.

```
Beginning scenario: RunUnsupportedScenario
Beginning scenario: RunBasicScenario_UnsafeRead
Process terminated. Illegal instruction: Attempted to execute an instruction code not defined by the processor.
   at System.RuntimeExceptionHelpers.FailFast(String, Exception, String, RhFailFastReason, IntPtr, IntPtr) + 0x231
   at System.RuntimeExceptionHelpers.GetRuntimeException(ExceptionIDs) + 0xb3
   at System.Runtime.EH.GetClasslibException(ExceptionIDs, IntPtr) + 0x34
   at JIT.HardwareIntrinsics.X86._Gfni.GfniAffineTest__GaloisFieldAffineTransformByte0.RunBasicScenario_UnsafeRead() + 0x38
   at JIT.HardwareIntrinsics.X86._Gfni.GfniAffineTest__GaloisFieldAffineTransformByte0.RunUnsupportedScenario() + 0x28
   at JIT.HardwareIntrinsics.X86._Gfni.Program.GaloisFieldAffineTransformByte0() + 0x30
   at Program.<<Main>$>g__TestExecutor37|0_38(StreamWriter tempLogSw, StreamWriter statsCsvSw, Program.<>c__DisplayClass0_0&) + 0x185
   at Program.<Main>$(String[] args) + 0x3fd
```

Cc @tannergooding

### PR DESCRIPTION
**Pull request by saucecontrol**
Fixes #110240 

GFNI is enabled opportunistically, and the HWIntrinsics tests check that the methods throw PNSE when not supported.  On NAOT, instead of throwing, the instructions are emitted, so they result in illegal instruction instead.

### CODE REVIEW

**Comment by saucecontrol** [comment]
For reference, the test that was failing:

https://github.com/dotnet/runtime/blob/d6eb35426ebdb09ee5c754aa9afb9ad6e96a3dec/src/tests/JIT/HardwareIntrinsics/X86/Shared/GfniAffineTest.template#L229-L248

This template is common across HWIntrinsics tests, and any of these will fail when an unsupported instruction set is allowed in NAOT.

**Comment by jkotas** [comment]
> This template is common across HWIntrinsics tests, and any of these will fail when an unsupported instruction set is allowed in NAOT.

The other tests dealt with the problem by getting disabled: https://github.com/dotnet/runtime/blob/main/src/tests/JIT/HardwareIntrinsics/X86/X86Serialize/Serialize_r.csproj#L6-L7 . Should we do the same here?

Ideally, the opportunistic set would contain all instructions by default. The only reason why it is not the case are JIT instruction encoding limitations.

**Comment by saucecontrol** [comment]
Got it, thanks.  I've only added the property for the base `Gfni` tests for now.  If the outerloop tests are run with `--instruction-set=avx` or higher, more things can fail.  Should we disable NAOT tests for anything that isn't expected to be supported in CI?

**Comment by MichalStrehovsky** [comment]
> Got it, thanks. I've only added the property for the base `Gfni` tests for now. If the outerloop tests are run with `--instruction-set=avx` or higher, more things can fail. Should we disable NAOT tests for anything that isn't expected to be supported in CI?

We don't really test with `--instruction-set`, except for a couple native AOT smoke tests. We'd need to do more work if we want to support this as a documented switch and that includes running more tests with these switches set (the switch is currently undocumented on learn.microsoft.com and we treat it as soft-documented).

**Comment by jkotas** [comment]
> This template is common across HWIntrinsics tests, and any of these will fail when an unsupported instruction set is allowed in NAOT.

It would be nice to fix this template to be compatible with native AOT: https://github.com/dotnet/runtime/issues/110293

File: src/tests/JIT/HardwareIntrinsics/X86/Gfni/Gfni_r.csproj [hunk 1]
```diff
@@ -2,6 +2,9 @@
   <PropertyGroup>
     <AssemblyName>X86_Gfni_r</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
+    <!-- Would have to translate Illegal instruction to PNSE -->
```
Comment (jkotas) [inline src/tests/JIT/HardwareIntrinsics/X86/Gfni/Gfni_r.csproj:6]:
```suggestion
    <!-- Would have to translate Illegal instruction to PNSE -->
    <!-- https://github.com/dotnet/runtime/issues/110293 -->
```

File: src/tests/JIT/HardwareIntrinsics/X86/Gfni/Gfni_ro.csproj [hunk 2]
```diff
@@ -2,6 +2,9 @@
   <PropertyGroup>
     <AssemblyName>X86_Gfni_ro</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
+    <!-- Would have to translate Illegal instruction to PNSE -->
```
Comment (jkotas) [inline src/tests/JIT/HardwareIntrinsics/X86/Gfni/Gfni_ro.csproj:6]:
```suggestion
    <!-- Would have to translate Illegal instruction to PNSE -->
    <!-- https://github.com/dotnet/runtime/issues/110293 -->
```

**Review by MichalStrehovsky — APPROVED** [review]

**Comment by MichalStrehovsky** [comment]
Thank you!

**Pull request merged**